public/coffee/ide/editor/EditorManager.coffee

```diff
@@ -121,11 +121,22 @@ define [
 
 		_bindToDocumentEvents: (doc, sharejs_doc) ->
 			sharejs_doc.on "error", (error, meta) =>
-				if error?.message?.match "maxDocLength"
+				if error?.message?
+					message = error.message
+				else if typeof error == "string"
+					message = error
+				else
+					message = ""
+				if message.match "maxDocLength"
 					@ide.showGenericMessageModal(
 						"Document Too Long"
 						"Sorry, this file is too long to be edited manually. Please upload it directly."
 					)
+				else if message.match "too many comments or tracked changes"
```

Comment:
Yea, I didn't like this, but it's what the doc updater currently returns. The error above, with a nicer message is emitted internally to the client, so we have more direct control. My default position was to not change out of laziness, but I will do if you ask me to.